R/zzz.R

```diff
@@ -1,9 +1,38 @@
 .onLoad <- function(...) {
+
   # register the extendr knitr chunk engine if knitr is available
   if (requireNamespace("knitr", quietly = TRUE)) {
     knitr::knit_engines$set(
       extendr = eng_extendr,
       extendrsrc = eng_extendrsrc
     )
   }
+
+  # Setting default options
+  # If rextendr options are already set, do not override
+  # NULL values are present for reference and may later be replaced
+  # by concrete values
+
+  rextendr_opts <- list(
+    # character scalar
+    # Controls default Rust toolchain; NULL corresponds to system's default
+    rextendr.toolchain = NULL,
+    # character vector
+    # Overrides Rust dependencies; mainly used for development
+    rextendr.patch.crates_io =  c(
+      'extendr-api = { git = "https://github.com/extendr/extendr" }',
+      'extendr-macros = { git = "https://github.com/extendr/extendr" }'
+    ),
+    # character scalar
+    # Default version of 'extendr_api' if no 'patch.crates_io' is specified
+    rextendr.extendr.version = "*",
+    # character scalar
+    # Default version of 'extendr_macros' if no 'patch.crates_io' is specified
+    rextendr.extendr_macros.version = "*"
```

Comment:
I'm a bit unclear about when to use `.` vs `_` as a separator but I guess it's fine the way you set it up.

Reply:
@clauswilke,
From what I observed in options present on my system and set by other packages -- `'.'` is used as a separator, e.g. `{ggplot2}` https://github.com/tidyverse/ggplot2/blob/9741da5050f81b7b5c012c56d02f45fc93d68f89/R/scale-colour.r#L43-L45.
So the naming convention looks like `<package>.<global_option>` and `<package>.<entity>.<property>`.
Usage of `'_'` is adopted from rextendr itself, where it replaces `'-'`, e.g. `patch.crates_io` is an equivalent of Cargo.toml's `[patch.crates-io]`. Same for `extendr-*`.